### ISSUE TITLE
'unicorn/no-this-assignment' seems redundant with ESLint's 'consistent-this'

### ISSUE THREAD

**Issue by Slapbox** [sindresorhus/eslint-plugin-unicorn#1499]
Is there some distinction I'm overlooking that necessitates `unicorn/no-this-assignment`? I assume yes, since the `consistent-this` rule has been available since [before ESLint v1](https://eslint.org/docs/rules/consistent-this#version).

**Comment by Slapbox** [comment]
Can anyone clarify how these rules differ? Five months later, I'm still not seeing it.

**Comment by fisker** [comment]
`consistent-this`  enforce consistent alias variable name, `unicorn/no-this-assignment` don't allow assign `this` to any variable.

```js
const that = this; // valid in `consistent-this` , invalid in `unicorn/no-this-assignment`
```
